### PR TITLE
Ignore SYNC_COMMITTEE_SUBNET_COUNT when parsing spec

### PR DESCRIPTION
**Pull request by ajsutton**
## PR Description
The constant `SYNC_COMMITTEE_SUBNET_COUNT` isn't in the actual config files but is in the REST API config response so we need to ignore it when parsing.

## Documentation

- [x] I thought about documentation and added the `documentation` label to this PR if updates are required.

## Changelog

- [x] I thought about adding a changelog entry, and added one if I deemed necessary.

### CODE REVIEW

File: data/provider/src/test/java/tech/pegasys/teku/api/ConfigProviderTest.java [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+import static org.assertj.core.api.Assertions.assertThat;
+
+import org.junit.jupiter.api.Test;
+import tech.pegasys.teku.api.response.v1.config.GetSpecResponse;
+import tech.pegasys.teku.spec.Spec;
+import tech.pegasys.teku.spec.SpecFactory;
+import tech.pegasys.teku.spec.config.SpecConfig;
+import tech.pegasys.teku.spec.config.SpecConfigLoader;
+
+class ConfigProviderTest {
```
Comment (regio) [inline data/provider/src/test/java/tech/pegasys/teku/api/ConfigProviderTest.java:25]: Should we add also a negative test when the config provider fail to load?

Comment (ajsutton) [reply data/provider/src/test/java/tech/pegasys/teku/api/ConfigProviderTest.java:25]: I'm not exactly sure what you mean. You can't start without having a `Spec` object so there shouldn't ever be a case where we can't generate the config response and there shouldn't ever be a case where we can't parse that output again.

We do have a bunch of tests in `SpecConfigReaderTest` for how it handles invalid configs - the aim here is just to check that the config we get from our REST API is considered valid.

Comment (regio) [reply data/provider/src/test/java/tech/pegasys/teku/api/ConfigProviderTest.java:25]: yes, I was thinking about the invalid configuration. I didn't realised this scenario was covered by SpecConfigReaderTest.

Comment (ajsutton) [reply data/provider/src/test/java/tech/pegasys/teku/api/ConfigProviderTest.java:25]: All good, it was a good question to ask.

**Review by regio — APPROVED** [review]
LGTM

**Pull request merged**